[WIP] Docs: Add reference entries for p5.strands hooks (feedback wanted)

Abhayaj247 opened the pull request:
This draft PR adds new JSDoc reference entries for a few p5.strands hooks, following the feedback in #7919.

- The new doc comments use JavaScript-only examples and describe the callback signature as discussed.
- I’ve included examples for :

- getWorldInputs
- combineColors
- getPointSize

- The doc comments are currently placed in a new file :

src/webgl/p5.strands.js

(Open to moving them if you prefer a different location.)

Could you please review the style, level of detail, and file location?

Once confirmed, I’ll update the rest of the hooks in the same format.

Thanks for your guidance!

lukeplowden commented on src/webgl/p5.strands.js:
```diff
@@ -0,0 +1,146 @@
+ * @function getWorldInputs
+ * @experimental
+ * @description
+ * Registers a callback to modify world-space vertex inputs for each vertex.
```
Maybe 'vertex inputs for each vertex' is a bit redundant here. I would remove the 'for each vertex' part or reword slightly

Abhayaj247 replied:
Reworded the description to remove redundancy and improve clarity.

```
// Before:
 * Registers a callback to modify world-space vertex inputs for each vertex.

// After:
 * Registers a callback to modify the world-space properties of each vertex in a shader.
 
 ```

lukeplowden commented on src/webgl/p5.strands.js:
```diff
@@ -0,0 +1,146 @@
+ * - {@link p5.baseStrokeShader}
+ *
+ * @param {function(inputs: { position: {x: number, y: number, z: number}, normal: {x: number, y: number, z: number}, texCoord: {x: number, y: number}, color: {r: number, g: number, b: number, a: number} }): object} callback
+ *        A function that receives the current inputs and returns the modified inputs.
```
Maybe something like "a callback function which receives and returns a Vertex struct."

lukeplowden commented on src/webgl/p5.strands.js:
```diff
@@ -0,0 +1,146 @@
+ * The callback receives the current point size (number) and should return the new size (number).
+ *
+ * This hook is available in:
+ * - {@link p5.pointShader}
```
Do we have a point shader?

Abhayaj247 replied:
Removed the reference to p5.pointShader and only listed the correct shaders for each hook.

```

// Before:
 * This hook is available in:
 * - {@link p5.pointShader}

// After:
 * This hook is available in:
 * - {@link p5.baseMaterialShader}
 * - {@link p5.baseNormalShader}
 * - {@link p5.baseColorShader}
 * - {@link p5.baseStrokeShader}
 
 ```

Abhayaj247 commented on src/webgl/p5.strands.js:
```diff
@@ -0,0 +1,146 @@
+ *   myShader = baseMaterialShader().modify(() => {
+ *     getWorldInputs((inputs) => {
+ *       // Move the vertex up and down in a wave
+ *       inputs.position.y += 20 * Math.sin(millis() * 0.001 + inputs.position.x * 0.05);
```
Updated the example to use sin() instead of Math.sin() as suggested.

```
// Before:
inputs.position.y += 20 * Math.sin(millis() * 0.001 + inputs.position.x * 0.05);

// After:
inputs.position.y += 20 * sin(millis() * 0.001 + inputs.position.x * 0.05);

```

Abhayaj247 commented:
**Thank you for the review and detailed feedback @lukeplowden .**

**I have made the following updates to address all the points :**

- Added more context and explanations for each hook, including where these functions can be used (e.g., inside baseMaterialShader().modify()) and an explanation of what “world space” means.

- Updated all code examples to use sin() instead of Math.sin().

- Reworded the phrase “vertex inputs for each vertex” for clarity and to avoid redundancy.

- Callback parameters are now described as receiving and returning a Vertex struct, and I have added a @typedef for Vertex at the top of the file.

- Removed the reference to p5.pointShader and only listed the correct shaders for each hook.

**You can see all these changes in my second commit in this PR.**

**Please let me know if there’s anything else I should update. Thank you!**

Abhayaj247 commented:
Hi @lukeplowden ,

I have addressed all the feedback and updated the documentation accordingly. If everything looks good and the changes are correct, please let me know so I can proceed with documenting the remaining hooks in the same style.

Or, if any further changes are needed, please let me know and I will update accordingly.

Thank you!

Abhayaj247 commented:
**Also, just to note, @lukeplowden : my previous commit was mistakenly deleted. All the updates mentioned above are now included in my current commit. Thank you for your understanding!**

Abhayaj247 commented:
Hi @lukeplowden , just following up on this PR.

I’ve made all the updates you suggested (details in my previous comment). Please let me know if there’s anything else I should update, or if this is good to go. Thank you!

Abhayaj247 commented:
> I also noticed that you added the reference documentation to the main branch, but p5.Strands is implemented in the dev-2.0 branch.
> 
> Could you please retarget your PR to the dev-2.0 branch and re-base your work on top of it?
> 
> Steps:
> 
> 1. In the PR, click Edit ➜ change the base from `main` to `dev-2.0` ➜ Save (Already I did)
> 
> Locally:
> 
> ```
> git fetch origin
> git checkout strands-docs-draft
> git rebase origin/dev-2.0
> git push --force-with-lease
> ```
> 
> This will make the PR show only your intended changes against `dev-2.0`, with conflicts resolved. Thanks!


Hi @perminder-17,
I’m following the steps to rebase my branch onto dev-2.0 as suggested :

```

git fetch origin
git checkout strands-docs-draft
git rebase origin/dev-2.0
git push --force-with-lease

```

**However, While rebasing onto dev-2.0 locally, I’m encountering repeated merge conflicts in files like README.md and .all-contributorsrc. Should I need resolve each conflict manually, or is it acceptable to skip these commits if they’re not relevant to my changes? Please advise on the preferred approach. Thank you!**

**Here’s an example of the error I’m seeing:**

For 

```
**For  .all-contributorsrc :**

CONFLICT (content): Merge conflict in .all-contributorsrc
error: could not apply 8b7f6f252... docs: update .all-contributorsrc [skip ci]
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 8b7f6f252... docs: update .all-contributorsrc [skip ci]

```


```
**For  README.md :**

CONFLICT (content): Merge conflict in README.md
error: could not apply ef7104d6e... docs: update README.md [skip ci]
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".       
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply ef7104d6e... docs: update README.md [skip ci]

```

perminder-17 commented:
> Just following up on my previous question regarding the repeated merge conflicts while rebasing onto dev-2.0 (specifically in files like README.md and .all-contributorsrc).

Would it be okay for you to open a new PR targeting `dev-2.0` branch if you're facing issues? The idea is not to solve the merge conflicts manually, but the easiest way for now would be to close this one and open a new PR copying the changes from this PR and pasting it to the `dev-2.0` ones?

Abhayaj247 commented:
**Closing this PR in favor of #7940, which ports these changes to the dev-2.0 branch as suggested by the maintainers. The new PR avoids the previous merge conflicts and follows the recommended workflow.**

Thank you to everyone who reviewed and provided feedback here!
